From 851281896e6d76b8dc6189352fcafa5e7590b65d Mon Sep 17 00:00:00 2001
From: Orlando Vazquez <ovazquez@gmail.com>
Date: Fri, 14 Dec 2018 10:00:53 -0800
Subject: [PATCH] TRITON-1034 Provisioning a VM with `do_not_inventory` set
 causes the workflow job to time out at putVm Reviewed by: Todd Whiteman
 <todd.whiteman@joyent.com> Approved by: Todd Whiteman
 <todd.whiteman@joyent.com>

---
 lib/workflows/job-common.js | 6 ++++++
 package.json                | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/workflows/job-common.js b/lib/workflows/job-common.js
index 0a91129..5b9f560 100644
--- a/lib/workflows/job-common.js
+++ b/lib/workflows/job-common.js
@@ -304,6 +304,12 @@ function pollTask(job, cb) {
 function putVm(job, cb) {
     var vmapi;
 
+    if (job.params.payload.do_not_inventory) {
+        cb(null,
+            'VM has do_not_inventory set, no need to putVm to VMAPI');
+        return;
+    }
+
     /*
      * Checks (polls) the state of a machine in VMAPI. It is used for provisions
      * and VM actions such as reboot and shutdown.
diff --git a/package.json b/package.json
index 95e3989..a4df7f3 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.6.5",
+  "version": "9.6.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

